app/tournament/match.js

```diff
@@ -0,0 +1,39 @@
+export class Match {
+    constructor(round) {
+        this.round = round;
+        this.players = [];
+        this.isFinished = false;
+    }
+
+    addPlayer(player) {
+        if (this.players.length >= 2) {
+            throw new Error("Match has already 2 players");
+        }
+        this.players.push(player);
+    }
+
+    setNextMatch(match) {
+        this.nextMatch = match;
+    }
+
+    setWinner(player) {
+        if (!this.isFinished) {
+            this.isFinished = true;
+            this.nextMatch.addPlayer(this.players[player]);
+        }
+    }
+}
+
+export class ChildMatch extends Match {
```

Comment:
Worst name in the world, unless you use ClassThatExtendsFromMatch. Use domain names instead.